### PR TITLE
Solsig indexed

### PR DESCRIPTION
**Pull request by peterbroadhurst**
After all the work in #71, I ended up missing one of the most important distinguishing things the feature was meant to help with 🤦 

The ABI signature of an event is identical for ERC-20/ERC-721, but some of the fields are indexed. I'd missed including that.

In PR chain with #73

### CODE REVIEW

File: pkg/abi/typecomponents.go [hunk 1]
```diff
@@ -371,13 +371,18 @@ func (tc *typeComponent) String() string {
 	}
 }
 
-func (tc *typeComponent) SolidityParamDef(inFunction bool) (string, []string) {
+func (tc *typeComponent) SolidityParamDef(isFunction, isEvent bool) (string, []string) {
 	isRef, paramDef, childStructs := tc.SolidityTypeDef()
-	if isRef && inFunction {
-		paramDef = fmt.Sprintf("%s memory", paramDef)
+	if isRef && isFunction {
+		paramDef += " memory"
```
Comment (EnriqueL8) [inline pkg/abi/typecomponents.go:377]: why the change from sprintF to `+=`?

**Review by EnriqueL8 — COMMENTED** [review]
a few comments

File: pkg/abi/abi.go [hunk 2]
```diff
@@ -736,7 +737,7 @@ func (e *Entry) SolidityDefCtx(ctx context.Context) (string, []string, error) {
 				if i > 0 {
 					buff.WriteString(", ")
 				}
-				s, childStructs, err := p.SolidityDefCtx(ctx, isFunction)
+				s, childStructs, err := p.SolidityDefCtx(ctx, true, false)
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/abi.go:740]: I moved away from that, because it's not accurate here... even when `isFunction` is true, the value to this should be `false` - I will rename `isFunction` to `isFunctionInputs`

[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/abi.go:740]: second guessing myself here - checking on qualifiers for return

[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/abi.go:740]: ok - yeah, modifiers (`memory`) valid on returns. So code is correct. happy to change for readability.

[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/abi.go:740]: Happy to change readability - changed the assurance of these values to comments

File: pkg/abi/typecomponents.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ type TypeComponent interface {
 	DecodeABIData(d []byte, offset int) (*ComponentValue, error)
 	DecodeABIDataCtx(ctx context.Context, d []byte, offest int) (*ComponentValue, error)
 
-	SolidityParamDef(inFunction bool) (solDef string, structDefs []string) // gives a string that can be used to define this param in solidity
+	SolidityParamDef(isFunction, isEvent bool) (solDef string, structDefs []string) // gives a string that can be used to define this param in solidity
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/typecomponents.go:69]: ok - moving interface to enum and refactoring

File: pkg/abi/abi.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply pkg/abi/abi.go:740]: closing as covered by enum request below

File: pkg/abi/typecomponents.go (hunk 1 above)
Comment (peterbroadhurst) [reply pkg/abi/typecomponents.go:377]: just felt more readable

**Review by EnriqueL8 — APPROVED** [review]
Thank for making those changes @peterbroadhurst , looks a lot nearer to me now!
